File: ios/ExpenseTracker/Domain/Expense.swift
import Foundation

struct Expense: Identifiable, Equatable {
    let id: UUID
    let title: String
    let amount: Decimal
    let createdAt: Date

    init(id: UUID = UUID(), title: String, amount: Decimal, createdAt: Date = Date()) {
        self.id = id
        self.title = title
        self.amount = amount
        self.createdAt = createdAt
    }
}

File: android/app/src/main/java/com/bkes994408/expensetracker/domain/Expense.kt
package com.bkes994408.expensetracker.domain

import java.math.BigDecimal
import java.time.Instant
import java.util.UUID

data class Expense(
    val id: UUID = UUID.randomUUID(),
    val title: String,
    val amount: BigDecimal,
    val createdAt: Instant = Instant.now(),
)
